packages/api/src/command/medical/document/document-query.ts

```diff
@@ -162,3 +179,26 @@ export const updateConversionProgress = async ({
     return updatedPatient;
   });
 };
+
+/**
+ * Returns the existing request ID if the previous query has not been entirely completed. Otherwise, returns a newly-generated request ID.
+ *
+ * @param docQueryProgress Progress of the previous query
+ * @returns uuidv7 string ID for the request
+ */
+function getOrGenerateRequestId(docQueryProgress: DocumentQueryProgress | undefined): string {
+  if (!docQueryProgress) return uuidv7();
+
+  const finishedDownload = docQueryProgress.download?.status === "completed";
```

Comment:
NABD, but I suggest we try to use `isXXX` to boolean constants, like `isDownloadFinished` - clearer to what it represents.

Reply:
Ah, gotcha